core/commands/storage.go

```diff
@@ -438,12 +448,78 @@ func getValidHost(retryQueue *storage.RetryQueue) (*storage.HostNode, error) {
 				return nil, err
 			}
 		} else {
+			// find peer
+			pi, err := findPeer(ctx, n, nextHost.Identity)
+			if err != nil {
+				// cannot find peer, remove from queue permanently
+				log.Error(err)
```

Comment:
It's common not able to find peer, shouldn't need so many logs

Reply:
if not find peer, I would offer back to queue and increase fail times.